### ISSUE TITLE
[BUG] Classification pre-trained model can make use of some serious refinement...

### ISSUE THREAD

**Issue by derei** [IacobIonut01/Gallery#549]
It's quite obvious from the attachments what I mean.
The worst is to not recognise Enterprise (blasphemy 😆).

![SmartSelect_20241208_114926_Gallery](https://github.com/user-attachments/assets/3129f162-4aeb-4a7e-a842-b13ad270c7ee)
![SmartSelect_20241208_114914_Gallery](https://github.com/user-attachments/assets/2ad512c4-ea50-488c-94d2-28a75fb6136c)
![SmartSelect_20241208_114859_Gallery](https://github.com/user-attachments/assets/94f9497e-5082-4ccf-8a60-552a7e50dce6)

**Comment by IacobIonut01** [comment]
This is the best I could find - which is the same as the Google Gallery Go uses.

The next update will be a major one where it would allow for manually creating and moving categories for the ones that do not match.


I'm not in any way an AI/Machine Learning expert nor I have any significant experience with it, but from my testing this is the best I could find that categories the media in a more general manner while being fast and efficient - which as you can see can definitely sacrifice the accuracy.

I have tried over 15 models - some were too specific (and very often wrong).

**Comment by IacobIonut01** [comment]
Also creating my own model is not a priority right now

**Comment by mon-jai** [comment]
@IacobIonut01 Just out of curiosity, what model are you using right now?

**Comment by IacobIonut01** [comment]
> @IacobIonut01 Just out of curiosity, what model are you using right now?

Mobile Ica 8bit with metadata

**Comment by newhinton** [comment]
@IacobIonut01 So this is entirely loca? I was surprised that it immediately started classifying my images.

**Comment by IacobIonut01** [comment]
> @IacobIonut01 So this is entirely loca? I was surprised that it immediately started classifying my images.

Yes, it uses Tensorflow Lite to process them in the background locally

**Comment by tdbe** [comment]
There are two major anachronistic dealbreakers here:

1.  You generated hundreds of useless categories, and I can't even search for them. So what's the point? There's also a lot of synonym categories like airplane aircraft aviation aerospace engineering (btw 0 of the images actually have anything to do with any of those words)

2. The accuracy is trash vs QuicPic or Tidy. I get one (1) swimwear image in Gallery, and and in picquery or tidy I have hundreds (accurate).

This will sound crass but, 'why don't you just use their code, it's right there': 
[https://github.com/greyovo/PicQuery](https://github.com/greyovo/PicQuery)
[https://github.com/slavabarkov/tidy](https://github.com/slavabarkov/tidy)

More importantly, you can actually SEARCH for the content instead of scroll through hundreds of uncalled for categories spam.

**Comment by IacobIonut01** [comment]
> There are two major anachronistic dealbreakers here:
> 
> 1.  You generate hundreds of useless categories, and I can't even search for them. So what's the point? There's also a lot of synonym categories like airplane aircraft aviation aerospace (btw 0 of the images actually have anything to do with any of those words)
> 
> 2. The accuracy is trash vs QuicPic or Tidy. I get one (1) swimwear image in Gallery, and and in picquery or tidy I have hundreds (accurate).
> 
> This will sound crass but, 'why don't you just use their code, it's right there and it's worked for years': 
> [https://github.com/greyovo/PicQuery](https://github.com/greyovo/PicQuery)
> [https://github.com/slavabarkov/tidy](https://github.com/slavabarkov/tidy)
> 
> More importantly, you can actually SEARCH for the content instead of scroll through hundreds of uncalled-for categories spam. And search is instant.

Thanks for the suggestions. 

"Why don't you just use their code?" - Simple, didn't know about those projects, but now I do thanks to you!

**Comment by IacobIonut01** [comment]
> There are two major anachronistic dealbreakers here:
> 
> 1. You generate hundreds of useless categories, and I can't even search for them. So what's the point? There's also a lot of synonym categories like airplane aircraft aviation aerospace (btw 0 of the images actually have anything to do with any of those words)
> 2. The accuracy is trash vs QuicPic or Tidy. I get one (1) swimwear image in Gallery, and and in picquery or tidy I have hundreds (accurate).
> 
> This will sound crass but, 'why don't you just use their code, it's right there and it's worked for years': https://github.com/greyovo/PicQuery https://github.com/slavabarkov/tidy
> 
> More importantly, you can actually SEARCH for the content instead of scroll through hundreds of uncalled-for categories spam. And search is instant.

Expect an update regarding this in the new Search overhaul update.

Currently I'm working on other features, but the Search overhaul will get prioritise right after that
